src/test/java/com/java8/wordsplay/FileReaderTest.java

```diff
@@ -102,32 +107,44 @@ void shouldReturnWordWithSortedEntries() throws Exception {
                 .collect(
                         groupingBy(
                                 (word -> word),
-                                Collectors.counting()
+                                counting()
                         )
                 )
                .entrySet()
                .stream()
-                . sorted( comparing(Entry::getKey)).collect(toList());
+               . sorted( comparing(Entry::getKey)).collect(toList());
 
         wordMap.forEach( entry1 -> System.out.println(entry1.getKey() + " " + entry1.getValue()));
 
     }
 
     @Test
     void shouldReturnWithNaturalOrder()  throws Exception {
+        final Supplier<TreeMap<String,Long>> naturalOrdering = () -> new TreeMap<String, Long>(Comparator.naturalOrder());
+
+        Map<String, Long> wordMapNaturalOrdering = getWordListComparingBy(naturalOrdering);
+        wordMapNaturalOrdering.forEach((key, value) -> System.out.println(key + " " + value));
+    }
+
+    @Test
+    void shouldReturnReverseOrder()  throws Exception {
+        final Supplier<TreeMap<String,Long>> reverseOrder = () -> new TreeMap<String, Long>(Comparator.reverseOrder());
+
+        Map<String, Long> wordMapReverseOrdering = getWordListComparingBy(reverseOrder);
+        wordMapReverseOrdering.forEach((key, value) -> System.out.println(key + " " + value));
+    }
+
 
-        Map<String, Long> wordMap =
-                wordList.stream()
-                        .collect(
-                                groupingBy(
-                                        (word -> word),
-                                        // creating natural order tree map dynamically
-                                        () -> new TreeMap<String, Long>(Comparator.naturalOrder()),
-                                        Collectors.counting()
-                                )
-                        );
-
-        wordMap.forEach((key, value) -> System.out.println(key + " " + value));
+    private TreeMap<String, Long> getWordListComparingBy(final Supplier<TreeMap<String, Long>> orderingSupplier) {
+        return wordList.stream()
+                .collect(
+                        groupingBy(
+                                (word -> word),
+                                orderingSupplier,
```

Comment:
does this line need orderingSupplier.get()?

Reply:
At the moment , the .get() is not required , as Supplier is present , though we can have a method with an Optional<Supplier> , like this - Optional<Supplier<TreeMap<String, Long>>> optionalOrderingSupplier